src/command_helpers.rs

```diff
@@ -301,14 +253,38 @@ pub(crate) fn command_add_output_file(
     }
 }
 
+#[cfg(feature = "parallel")]
+fn forward_available_stderr_as_warnings(
+    child_stderr_reader: &mut Option<BufReader<std::process::ChildStderr>>,
+) {
+    if let Some(child_stderr_reader) = child_stderr_reader.as_mut() {
+        if let Ok(available) = child_stderr_reader.fill_buf() {
+            let stdout = io::stdout();
+            let mut consumed = 0;
+            for line in available.split(|&b| b == b'\n') {
```

Comment:
I think we need to handle cases where `available` does not include a newline and there's more data to come.

In this case, I think you should keep the data until newline or eof, because if you prints part of the line as a warning, then you would be adding newline to the originaly message.

Reply:
Ah, yep, I forgot that `split` will provide the rest of the items in last iteration.